api/agent/agent.go

```diff
@@ -652,6 +668,20 @@ func (c *container) WriteStat(ctx context.Context, stat drivers.Stat) {
 	for key, value := range stat.Metrics {
 		span.LogFields(log.Uint64("fn_"+key, value))
 	}
+	// Fn polls for stats as long as container lives,
+	// so we have to write stats only once
+	if c.callID != "" {
+		err := c.ds.UpdateCallMetrics(ctx,
```

Comment:
I think this present layering issues here, as well as being untenable from a db perspective -- this is too many writes to the db. we should aggregate stats inside of `WriteStat` and then write them to the db once inside of `call.End` -- so aside from moving this function and changing it's contract to write a list of stats, `WriteStat` will need to accumulate data points somewhere.